test/ci_tests/header_check.py

```diff
@@ -98,7 +101,43 @@ def get_top_comments(_data):
     return comments
 
 
+def get_committed_files():
+    """
+    Get a list of files that are part of the current commit
+    """
+    git_executable = shutil.which("git")
+    if not git_executable:
+        raise RuntimeError("git executable not found in PATH")
+    result = subprocess.run(
+        [git_executable, "diff", "--name-only", "--cached"],  # noqa: S603
```

Comment:
This only checks changed files in the git tree right? That's even better to reduce the per commit runtime cost!